torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -39,18 +50,57 @@ def regenerate_view(self, bases_list: List[Tensor]):
         if not self.is_view:
             return bases_list[self.base_index]
 
+        if self.slice_info is not None:
+            return torch.ops.aten.slice.Tensor(
+                bases_list[self.base_index], *self.slice_info
+            )
+
+        return torch.as_strided(
+            bases_list[self.base_index],
+            *self.as_strided_info,
+        )
+
         assert self.stride is not None
         assert self.size is not None
         assert self.storage_offset is not None
 
         return torch.as_strided(
             bases_list[self.base_index],
-            self.size,
-            self.stride,
-            self.storage_offset,
+            *self.as_strided_info,
         )
 
 
+# return None or (dim, start, end)
+def try_use_slice(base, tensor):
```

Comment:
I'm concerned about what happens with dynamic shapes. As I mentioned in the other PR better testing would help convince me that it works out

Reply:
check test_recompiles added it in the previous should address this comment.